app/models/concerns/tagable.rb

```diff
@@ -6,17 +6,36 @@ module Tagable
     has_many :tags, through: :taggings
   end
 
-  TAGABLE_CLASSES = [Entity, List, Relationship]
+  class_methods do
+    def category_str
+      name.downcase.pluralize
+    end
+
+    def category_sym
+      category_str.to_sym
+    end
+  end
 
-  # sorts a list of tagables in descending order of relationships to tagables w/ same tag
-  def self.sort_by_related_tagables(tagables)
-    tagables
+  # () -> Array[ClassConstant]
+  def self.classes
+    @classes ||= [Entity, List, Relationship].freeze
   end
 
-  def self.page_param_of(klass_name)
-    (klass_name.to_s.downcase + "_page").to_sym
+  # () -> Array[Symbol]
+  def self.categories
+    @categories ||= classes.map(&:category_sym).freeze
   end
 
+  # str|sym -> ClassConstant
+  def self.class_of(category)
+    category.to_s.singularize.classify.constantize
+  end
+
+  # NOTE(@aguestuser): this constant *cannot* go before `included` and `class_methods` blocks
```

Comment:
because of loading?

Reply:
yes! it was because of loading! but also: that was true when it was a constant. but it's not a constant anymore!!! probably can go wherever. lemme run an experiment to see. (and update comment or remove it accordingly.)